### PR TITLE
Myhentaicomics ripper now rips all pages from tag and searches

### PR DESCRIPTION
**Pull request by cyian-1756**
I haven't completely tested this yet but it should work 

test links 

http://myhentaicomics.com/index.php/Danny-Phantom-An-Erotic-Parody

http://myhentaicomics.com/index.php/search?q=lemay

http://myhentaicomics.com/index.php/tag/323/Artist%3A+James+Lemay

### CODE REVIEW

**Comment by rautamiekka** [comment]
If you're testing for the address starting with `http`, do you want only `http` or is `https` fine as well ? The comments don't tell that. If not, you could use RegEx to make the `s` not allowed, or look for `http:` instead, which is faster and easier than RegEx anyway, and makes sure you get just HTTP if only HTTP is what you want. If the string contains, or should, just that anyway, 'starts with' checking is kinda wrong IMO.

Also, didn't Java, like Python, support boolean truth matching by just `!variable` instead of `variable == false` ?

**Comment by cyian-1756** [comment]
@rautamiekka 

> If you're testing for the address starting with http, do you want only http or is https fine as well ? 

Do you mean this (`boolean b = imageSource.startsWith("http");
`) line? Because that should be only http. It's to stop the ripper from picking up the site logo (Which is the only image link to start with http on the page)

> Also, didn't Java, like Python, support boolean truth matching by just !variable instead of variable == false ?

Yes it does, but I didn't know that when I first wrote this ripper. I will fix it soon.

Sorry if I misunderstood anything in your comment but I'm a bit out of it right now

**Comment by rautamiekka** [comment]
^ Nah, you got it. When you're incompletely matching a string for something like this, try to match as much as possible to hopefully minimize introducing sneaky bugs, but I know from multiple first-hand experiences that's hard.

File: src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java [hunk 1]
```diff
@@ -122,53 +123,134 @@ public Document getNextPage(Document doc) throws IOException {
...
+            String link = elem.attr("href");
+            logger.info("Grabbing album " + link);
+            pagesToRip = getNextAlbumPage(link);
+            logger.info(pagesToRip);
+            for (String element : pagesToRip) {
+                Document album_doc;
+                try {
+                    logger.info("grabbing " + element + " with jsoup");
+                    boolean startsWithhttp = element.startsWith("http://");
+                    if (startsWithhttp == false) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java]: !startsWithhttp 

nitpick: also capitalize Http in the var name

File: src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java [hunk 2]
```diff
@@ -122,53 +123,134 @@ public Document getNextPage(Document doc) throws IOException {
...
+                } catch(IOException e){
+                    logger.warn("Failed to log link in Jsoup");
+                    album_doc = null;
+                    e.printStackTrace();
+                }
+                for (Element el :album_doc.select("img")) {
+                    String imageSource = el.attr("src");
+                    // This bool is here so we don't try and download the site logo
+                    boolean b = imageSource.startsWith("http://");
+                    if (b == false) {
```
Comment (metaprime) [inline src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java]: !b

But honestly I don't think you need the boolean var if you're only going to use it in this one test. You can just inline it.

File: src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java (hunk 1 above)
Comment (cyian-1756) [inline src/main/java/com/rarchives/ripme/ripper/rippers/MyhentaicomicsRipper.java]: Fixed in my latest push

**Comment by metaprime** [comment]
One more thing before I can merge, could you please merge from master and resolve the merge conflict?

**Comment by cyian-1756** [comment]
@metaprime I've merged from master (I think) and fixed the conflict

**Pull request merged**